src/libraries/Microsoft.PowerFx.Core/Functions/Delegation/DelegationStrategies/DelegationValidationStrategy.cs

```diff
@@ -424,45 +424,19 @@ protected virtual bool IsValidAsyncOrImpureNode(TexlNode node, TexlBinding bindi
             Contracts.AssertValue(binding);
 
             var isAsync = binding.IsAsync(node);
-            var isPure = binding.IsPure(node);
 
-            if (!isAsync && isPure)
-            {
-                return true;
-            }
-
-            // Async predicates and impure nodes are not supported unless Features say otherwise.
             // Let CallNodes for delegatable async functions be marked as being Valid to allow
             // expressions with delegatable async function calls to be delegated
 
-            // Impure nodes should only be marked valid when Feature is enabled.
-            if (!isPure && !binding.Features.AllowImpureNodeDelegation)
+            if (!isAsync)
             {
-                TrackingProvider.Instance.SetDelegationTrackerStatus(DelegationStatus.ImpureNode, node, binding, trackingFunction ?? Function, DelegationTelemetryInfo.CreateImpureNodeTelemetryInfo(node, binding));
+                return true;
             }
             else
             {
-                if (!isAsync)
-                {
-                    return true;
-                }
-                else if (binding.Features.AllowAsyncDelegation)
-                {
-                    // If the feature is enabled, enable delegation for
-                    // async call, first name and dotted name nodes.
-                    return (node is CallNode) || (node is FirstNameNode) || (node is DottedNameNode);
-                }
-            }
-
-            if (isAsync)
-            {
-                TrackingProvider.Instance.SetDelegationTrackerStatus(DelegationStatus.AsyncPredicate, node, binding, trackingFunction ?? Function, DelegationTelemetryInfo.CreateAsyncNodeTelemetryInfo(node, binding));
+                // Enable delegation for async call, first name, and dotted name nodes.
+                return (node is CallNode) || (node is FirstNameNode) || (node is DottedNameNode);
```

Comment:
should we have a call to SuggestDelegationHintAndAddTelemetryMessage whenever this returns false?

Reply:
If we cannot delegate something, we should emit a warning

Reply:
Actually nevermind, the call to the delegation hint/telemetry wasn't being called before since binding.Features.AllowAsyncDelegation has been true for over a year